diff --git a/images/kernel/config-5.15.93 b/images/kernel/config-5.15.93
index 914e265..aaac59a 100644
--- a/images/kernel/config-5.15.93
+++ b/images/kernel/config-5.15.93
@@ -2,9 +2,9 @@
 # Automatically generated file; DO NOT EDIT.
 # Linux/x86 5.15.93 Kernel Configuration
 #
-CONFIG_CC_VERSION_TEXT="gcc (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0"
+CONFIG_CC_VERSION_TEXT="gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0"
 CONFIG_CC_IS_GCC=y
-CONFIG_GCC_VERSION=110300
+CONFIG_GCC_VERSION=110400
 CONFIG_CLANG_VERSION=0
 CONFIG_AS_IS_GNU=y
 CONFIG_AS_VERSION=23800
@@ -890,7 +890,7 @@ CONFIG_NET_PTP_CLASSIFY=y
 # CONFIG_MPLS is not set
 # CONFIG_NET_NSH is not set
 # CONFIG_HSR is not set
-# CONFIG_NET_SWITCHDEV is not set
+CONFIG_NET_SWITCHDEV=y
 # CONFIG_NET_L3_MASTER_DEV is not set
 # CONFIG_QRTR is not set
 # CONFIG_NET_NCSI is not set
@@ -930,6 +930,7 @@ CONFIG_FIB_RULES=y
 # CONFIG_NET_IFE is not set
 # CONFIG_LWTUNNEL is not set
 CONFIG_NET_SELFTESTS=y
+CONFIG_NET_DEVLINK=y
 CONFIG_FAILOVER=y
 CONFIG_ETHTOOL_NETLINK=y
 
@@ -1536,7 +1537,7 @@ CONFIG_MDIO_DEVRES=y
 
 # CONFIG_VMXNET3 is not set
 # CONFIG_FUJITSU_ES is not set
-# CONFIG_NETDEVSIM is not set
+CONFIG_NETDEVSIM=y
 CONFIG_NET_FAILOVER=y
 # CONFIG_ISDN is not set
 
@@ -3136,4 +3137,4 @@ CONFIG_CC_HAS_SANCOV_TRACE_PC=y
 CONFIG_ARCH_USE_MEMTEST=y
 # CONFIG_MEMTEST is not set
 # end of Kernel Testing and Coverage
-# end of Kernel hacking
+# end of Kernel hacking
\ No newline at end of file
